pkg/apis/common/v1alpha1/object_interfaces.go

```diff
@@ -41,6 +41,11 @@ type Reconcilable interface {
 	GetStatusManager() *StatusManager
 }
 
+type AdapterConfigurable interface {
```

Comment:
It would be nice to have a comment on that interface, like in the suggestion below, or anything else that conveys what this interface is useful for.

```suggestion
// AdapterConfigurable is implemented by types that can override the default
// configuration of their receive adapter.
type AdapterConfigurable interface {
```